### PR TITLE
feat: support storing page ordering of folders in DB

### PR DESCRIPTION
**Pull request by dcshzj**
## Problem

We need to support page ordering within folders, but there is no way to represent this inside Studio.

## Solution

**Breaking Changes**

- [ ] Yes - this PR contains breaking changes
- [x] No - this PR is backwards compatible

**Features**:

- Add support for storing the page ordering of folders inside the `_pages` permalink with a new `PageOrder` resource type.
- Prevent this new `PageOrder` resource from appearing in the list of pages when viewing a folder inside Studio.
- Updated the publisher script to take the page ordering inside this `PageOrder` resource into account.

**Notes**:

- We are not adding support for editing this page ordering yet, this has to be manually done via editing the database.

### CODE REVIEW

File: apps/studio/prisma/generated/generatedEnums.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ export const ResourceType = {
   CollectionLink: "CollectionLink",
   CollectionPage: "CollectionPage",
   IndexPage: "IndexPage",
+  PageOrder: "PageOrder",
```
Comment (harishv7) [inline apps/studio/prisma/generated/generatedEnums.ts:14]: Could _pages be used for potentially storing other page related metadata? If so, using a more generic name like PageMeta or PageSettings would be good

File: tooling/build/scripts/publishing/index.ts [hunk 2]
```diff
@@ -139,6 +146,8 @@ async function main() {
       }
     }
 
+    logDebug("Sitemap entries:", sitemapEntries)
```
Comment (harishv7) [inline tooling/build/scripts/publishing/index.ts:149]: is this to be removed?

File: tooling/build/scripts/publishing/index.ts [hunk 3]
```diff
@@ -159,7 +168,7 @@ async function main() {
       ),
     }
 
-    logDebug("Intermediate sitemap:", sitemap)
+    logDebug("Intermediate sitemap:", JSON.stringify(sitemap, null, 2))
```
Comment (harishv7) [inline tooling/build/scripts/publishing/index.ts:171]: same here

File: apps/studio/src/utils/sitemap.ts [hunk 4]
```diff
@@ -23,14 +23,19 @@ const getSitemapTreeFromArray = (
...
+        resource.type !== "PageOrder"
+      )
     }
     return (
       resource.parentId === parentId &&
       resource.permalink !== INDEX_PAGE_PERMALINK
     )
   })
 
+  // TODO: Sort the children by the page ordering if the PageOrder resource exists
```
Comment (harishv7) [inline apps/studio/src/utils/sitemap.ts:38]: does the order in the sitemap matter since we are already sorting this during publishing stage?

File: apps/studio/public/assets/css/preview-tw.css [hunk 5]
```diff
@@ -4133,11 +4129,29 @@ video {
   border-color: rgb(247 143 30 / var(--tw-border-opacity));
 }
 
+.focus\:bg-utility-highlight:where([data-rac])[data-focused] {
```
Comment (harishv7) [inline apps/studio/public/assets/css/preview-tw.css:4132]: are these css changes related to this page order?

File: apps/studio/prisma/generated/generatedEnums.ts (hunk 1 above)
Comment (dcshzj) [reply apps/studio/prisma/generated/generatedEnums.ts:14]: Hmm no, page-related metadata are stored within those individual pages. This is a page ordering that is tied to the folder and not to individual pages.

File: apps/studio/public/assets/css/preview-tw.css (hunk 5 above)
Comment (dcshzj) [reply apps/studio/public/assets/css/preview-tw.css:4132]: No, I think might have been previous changes that went in without doing the build step.

File: apps/studio/src/utils/sitemap.ts (hunk 4 above)
Comment (dcshzj) [reply apps/studio/src/utils/sitemap.ts:38]: This is used for the preview only, so we are maintaining the same behaviour as much as possible with what the user will see on the actual site.

File: tooling/build/scripts/publishing/index.ts (hunk 2 above)
Comment (dcshzj) [reply tooling/build/scripts/publishing/index.ts:149]: This is `logDebug`, so it only appears if `DEBUG` is set to `true`. Keeping this so we can have a way to understand how the sitemap looks like during the build process.

File: tooling/build/scripts/publishing/index.ts (hunk 3 above)
Comment (dcshzj) [reply tooling/build/scripts/publishing/index.ts:171]: Same as above.

File: apps/studio/prisma/generated/generatedEnums.ts (hunk 1 above)
Comment (harishv7) [reply apps/studio/prisma/generated/generatedEnums.ts:14]: Hmm understand, I just meant if there's potentially other use cases for _pages in future in which using a generic name now will help prevent migrations etc

**Review by harishv7 — APPROVED** [review]
lgtm, with one main comment on the naming of the resource type to be more generic. leave it to @dcshzj to take a call on this

File: apps/studio/prisma/generated/generatedEnums.ts (hunk 1 above)
Comment (dcshzj) [reply apps/studio/prisma/generated/generatedEnums.ts]: Renamed PageOrder to FolderMeta and `_pages` to `_meta`. I think we can have the option to have a CollectionMeta type in the future to control the ordering of items within the collection, but will need to be careful about the difference between this folder ordering meta vs normal folder metadata.
